include/display.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = 0,
+
+	PIXEL_FORMAT_END,
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(struct device *dev);
+
+/**
+ * @typedef display_off_api
+ * @brief Callback API to turn display off
+ * See display_off() for argument description
+ */
+typedef int (*display_off_api)(struct device *dev);
+
+/**
+ * @typedef display_write_bitmap_api
+ * @brief Callback API for writing bitmap
+ * See display_write_bitmap() for argument description
+ */
+typedef int (*display_write_bitmap_api)(
+	const struct device *dev, const u16_t x, const u16_t y, const u16_t w,
+	const u16_t h, const enum display_pixel_format pixel_format,
+	const void *data);
+
+/**
+ * @typedef display_is_pixel_format_supported_api
+ * @brief Callback API to check if pixel format is supported by display
+ * See display_is_pixel_format_supported() for argument description
+ */
+typedef bool (*display_is_pixel_format_supported_api)(
+	const struct device *dev, const enum display_pixel_format pixel_format);
+
+/**
+ * @brief Display driver API
+ * API which a display driver should expose
+ */
+struct display_driver_api {
```

Comment:
It lacks basic functions to query display resolution (or capabilities) and set contrast.

Reply:
I will add a function to set the contrast

Reply:
Function to set contrast and query capabilities added to API